src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -434,6 +436,38 @@ private void GenerateClassMarshals(Class @class)
             NewLine();
         }
 
+         private bool CheckClassIsStructible(Class @class)
+         {
+             if(!@class.DeclaredStruct)
+                 return false;
+
+             if (@class.IsValueType)
+                 return true;
+             if (@class.IsInterface)
+                 return false;
+             if (@class.IsStatic)
+                 return false;
+             if (@class.IsAbstract)
+                 return false;
+             if (@class.HasBase)
+                 return false;
+             if (@class.Methods.Any(m => m.IsStatic))
```

Comment:
C# structs are allowed to have static members and operators so this fix is incorrect. If the only problem with those tests is their static members, there's some other bug that needs fixing.